_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,256 @@
+
+## A few words about faasd
+
+OpenFaaS was created to have the freedom to run serverless functions anywhere, be it on a Raspberry Pi, in a data center, or the cloud. This started with faas-swarm to run on Docker Swarm and faas-netes to run on Kubernetes. faas-netes is the current recommendation for running OpenFaaS in production.
+Though, it's not necessary that you might want or even infact need Kubernetes, especially when considering the complexities involved with managing a Kubernetes cluster, and that's how faasd was born.
+faasd is just like OpenFaaS with the same ecosystem but it spares you from the complexity involved that comes with mangaging a distributed system.
+
+In this tutorial, we'll see how we can get OpenFaaS functions up and running on a faasd appliance with Github Actions.
+
+
+## Prerequisites
+
+* A VM exposed to the Internet
+
+For this tutorial we'll be using a [DigitalOcean](https://digitalocean.com/) Droplet.
+
+* faas-cli
+
+faas-cli is a command line tool used to interact with OpenFaaS, from deploying your functions to viewing logs.
+Follow the installation instructions [here](https://docs.openfaas.com/cli/install/)
+
+> You can follow along with the code used in this tutorial from this [repo](https://github.com/utsavanand2/faasd-hello)
+
+## Create a Droplet with faasd
+
+Create a DigitalOcean Droplet with a Ubuntu distribution, with any plan according to what you want to run, faasd being light weight I'm going to use the $5 basic plan.
+
+![Choose a distribution and a plan](/images/2020-09-19-github-actions-faasd/choose-distro.jpg)
+
+Under `Select additional options`, select `User data` and paste in the contents of the cloud-config file from [here](https://github.com/openfaas/faasd/blob/master/cloud-config.txt)
+
+![Select additional options](/images/2020-09-19-github-actions-faasd/additional-options-cloud-config.jpg)
+
+
+![faasd droplet](/images/2020-09-19-github-actions-faasd/faasd-droplet.jpg)
+
+Within as little as 30 seconds, your faasd instance will be up and running.
+
+## Get the OpenFaaS gateway password
+
+Get the gateway password by running the following commands
+
+```sh
+export IP="143.110.184.36"
+export PASSWORD=$(ssh root@$IP "sudo cat /var/lib/faasd/secrets/basic-auth-password")
+```
+
+Assuming that you have faas-cli already installed, login to faasd with faas-cli with the following commands
+
+```sh
+export OPENFAAS_URL=http://$IP:8080
+echo $PASSWORD | faas-cli login --password-stdin -u admin -g $OPENFAAS_URL
+```
+
+## Create an OpenFaaS function
+
+Now that we've logged into faasd with faas-cli, we're ready to run our functions.
+I'm going to deploy a golang function for this tutorial.
+
+```sh
+# create a new directory for our project
+mkdir faasd-hello
+# pull golang-http templates
+faas-cli template store pull golang-http
+# create a new function with faas-cli
+faas-cli new faasd-hello --lang golang-http --prefix utsavanand2
```

Comment:
Is `--prefix` needed if the build uses shrinkwrap?

Reply:
Yes! That is redundant.